amida-tech/greyscale
# Bug/inba 896 jf foreign constraints

#### What does this PR do?
Adds a foreign constraint to the `ProjectUsers` and `ProjectUserGroups` table. Also resolve an issue which may have been leaving orphan entries in `ProjectUserGroups`. 

#### Related JIRA tickets:
INBA-896

#### How should this be manually tested?
**Note:** Make sure your `DOMAIN`, `AUTH_SERVICE_URL`, `SURVEY_SERVICE_URL`, `MESSAGE_SERVICE_URL` variables (this being pre-DEVOPS-207) do not have a slash at the end. So...

INDABA_CLIENT_URL=http://localhost:3000
DOMAIN=http://localhost:3005
AUTH_SERVICE_URL=http://localhost:4000/api/v1
SURVEY_SERVICE_URL=http://localhost:9005/api/v1.0
MESSAGE_SERVICE_URL=http://localhost:4001/api/v1

To test, there are two ways to update your database. If you are working on something and don't want to smash your data, look in `greyscale/backend/db_setup/migration` for a script you can run to update your postgres database manually. 

If you can, please blow away your database and run `node seed.js` to reseed your database and ensure that `ProjectUsers` and `ProjectUserGroups` have two foreign keys for their ids and primary keys. 

Open up the tables for `ProjectUsers`, `ProjectUserGroups`, WorkflowStepGroups` and `UserGroups`. 
Go into a project on Indaba and create a user group, put users into it and assign it to the workflow.
Check those four opened tables to see the new values.
Then in Indaba, delete the newly created group (you should get various warnings about the project being active and it being assigned to the workflow, just agree). Ensure that the new entries are wiped out for all four tables.

#### Background/Context

#### Screenshots (if appropriate):
